RiotNSE/NotificationService.swift

```diff
@@ -345,69 +351,107 @@ class NotificationService: UNNotificationServiceExtension {
-                                    case kMXMessageTypeText:
-                                        notificationBody = messageContent
-                                        break
-                                    case kMXMessageTypeEmote:
-                                        notificationBody = NSString.localizedUserNotificationString(forKey: "ACTION_FROM_USER", arguments: [eventSenderName as Any, messageContent as Any])
-                                        break
-                                    case kMXMessageTypeImage:
-                                        notificationBody = NSString.localizedUserNotificationString(forKey: "IMAGE_FROM_USER", arguments: [eventSenderName as Any, messageContent as Any])
-                                        break
-                                    default:
-                                        // Encrypted messages falls here
-                                        notificationBody = NSString.localizedUserNotificationString(forKey: "MESSAGE", arguments: [])
-                                        break
+                            case kMXMessageTypeFile:
+                                notificationBody = NSString.localizedUserNotificationString(forKey: "FILE_FROM_USER", arguments: [eventSenderName, messageContent as Any])
+                            
+                            // All other message types such as text, notice, server notice etc
+                            default:
+                                if event.isReply() {
+                                    let parser = MXReplyEventParser()
+                                    let replyParts = parser.parse(event)
+                                    notificationBody = replyParts.bodyParts.replyText
+                                } else {
+                                    notificationBody = messageContent
                                 }
                             }
-                            break
                         case .roomMember:
-                            if roomDisplayName != nil && roomDisplayName != eventSenderName {
-                                notificationBody = NSString.localizedUserNotificationString(forKey: "USER_INVITE_TO_NAMED_ROOM", arguments: [eventSenderName as Any, roomDisplayName as Any])
+                            // If the current user is already joined, display updated displayname/avatar events.
+                            // This is an unexpected path, but has been seen in some circumstances.
+                            if NotificationService.backgroundSyncService.roomSummary(forRoomId: roomId)?.membership == .join {
+                                notificationTitle = self.messageTitle(for: eventSenderName, in: roomDisplayName)
+                                
+                                // If the sender's membership is join and hasn't changed.
+                                if event.content["membership"] as? String == "join",
```

Comment:
You can use `MXRoomMemberEventContent` model to parse the event content. And `kMXMembershipStringJoin` to compare strings as well.